code/game/objects/items/devices/traitordevices.dm

```diff
@@ -1,22 +1,35 @@
 /obj/item/jammer
 	name = "radio jammer"
-	desc = "Device used to disrupt nearby radio communication."
+	desc = "Fog of war that fits your pocket. Flicking the switch and extending the antenna will scramble nearby radio comms, making outgoing messages hard to understand."
 	icon = 'icons/obj/device.dmi'
 	icon_state = "jammer"
+	item_state = "jammer"
+	w_class = WEIGHT_CLASS_TINY
+	actions_types = list(/datum/action/item_action/toggle_radio_jammer)
 	var/active = FALSE
 	var/range = 12
 
 /obj/item/jammer/Destroy()
 	GLOB.active_jammers -= src
 	return ..()
 
+/obj/item/jammer/update_icon_state()
+	if(active)
+		icon_state = "[initial(icon_state)]-on"
+	else
+		icon_state = "[initial(icon_state)]"
+
 /obj/item/jammer/attack_self(mob/user)
-	to_chat(user, "<span class='notice'>You [active ? "deactivate" : "activate"] [src].</span>")
+	to_chat(user, "<span class='notice'>You [active ? "deactivate" : "activate"] [src]. [active ? "It goes quiet with a small click." : "It starts to hum silently."] </span>")
```

Comment:
This is a bit awkward with src being surrounded by two of the same checks. Is there any way for this to be one `active ?` check?

Reply:
i did this